readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -111,7 +210,6 @@ def test_incremental_state_update_with_no_update(self):
                 self.mocks.mocks['api_v2.build']().put.assert_called_with({
                     'id': DUMMY_BUILD_ID,
                     'version': self.version.pk,
-                    'success': True,
```

Comment:
Mmm...

Why it was passing before you remove this line?

I think that we have to check here that _after_ we exit the `with` context, we have done a call to `put` with the `sucess=True` in the first `build_env` and we didn't call it in the second one (`update_on_success=False`)

Reply:
https://github.com/rtfd/readthedocs.org/pull/3764#discussion_r173648808

Was because the context manager capture the assert exception to exit the context gracefully.

https://github.com/rtfd/readthedocs.org/blob/c5d102a4b119e191fd6a7ea201c776e84dd177f8/readthedocs/doc_builder/environments.py#L436-L445 

So, now I'm checking if the context manager is _exceptions free_ https://github.com/rtfd/readthedocs.org/pull/3764/files#diff-6d379045f32beb719b67ece38d86bf74R222